packages/components/src/Interactive.ts

```diff
@@ -35,124 +35,124 @@ const allowedHoverVariables = [
   'boxShadow',
   'opacity',
   'textDecoration',
-];
+] as const;
+
+// Union type of all values in allowedHoverVariables.
+type HoverStyleProperty = (typeof allowedHoverVariables)[number];
+
 const pointerTags = ['button', 'area', 'a', 'select'];
 const pointerTypes = ['radio', 'checkbox', 'submit', 'button'];
 
-export type InteractiveProps = {
-  children?: ReactNode;
-  className?: string;
-  disabled?: boolean;
-  hover?: React.CSSProperties;
-  role?: string;
-  style?: React.CSSProperties;
-  tabIndex?: number;
+// Since Interactive may render any HTML element based on the tagName prop,
+// use AllHTMLAttributes<HTMLElement> to allow this to accept attributes
+// for any HTML element.
+// TODO: Can a more specific type be used by parameterizing this with a tag name?
```

Comment:
I had tried this at one point (https://github.com/DataBiosphere/terra-ui/pull/3491/files#diff-4a807bcdfee92ec370e7587e55d86099290b729c7566d618792772fd7981e796) but we kept running into type errors and having to resort to casting props objects.

It _might_ work better now that these components are native TS instead of having types defined in `.d.ts` files, but that's a project for another day.